sqlmesh/core/snapshot/definition.py

```diff
@@ -1642,6 +1642,33 @@ def missing_intervals(
     return missing
 
 
+@lru_cache(maxsize=None)
+def expand_range(start_ts: int, end_ts: int, interval_unit: IntervalUnit) -> t.List[int]:
+    croniter = interval_unit.croniter(start_ts)
+    timestamps = [start_ts]
+
+    while True:
+        ts = to_timestamp(croniter.get_next(estimate=True))
+
+        if ts > end_ts:
+            if len(timestamps) > 1:
+                timestamps[-1] = end_ts
+            else:
+                timestamps.append(end_ts)
+            break
+
+        timestamps.append(ts)
+    return timestamps
+
+
+@lru_cache(maxsize=None)
+def expand_range_as_interval(
```

Comment:
this can just be done inline in scheduler and doesn't need to be cached, the expensive part is croniter